irohad/ametsuchi/impl/postgres_wsv_query.cpp

```diff
@@ -88,23 +85,28 @@ namespace iroha {
       };
     }
 
-    nonstd::optional<model::Account> PostgresWsvQuery::getAccount(
+    nonstd::optional<std::shared_ptr<Account>> PostgresWsvQuery::getAccount(
         const std::string &account_id) {
       return execute_("SELECT * FROM account WHERE account_id = "
-                      + transaction_.quote(account_id)
-                      + ";")
-                 | [&](const auto &result) -> nonstd::optional<model::Account> {
+                      + transaction_.quote(account_id) + ";")
+                 | [&](const auto &result)
+                 -> nonstd::optional<std::shared_ptr<Account>> {
         if (result.empty()) {
           log_->info(kAccountNotFound, account_id);
           return nonstd::nullopt;
         }
-        model::Account account;
-        auto row = result.at(0);
-        row.at(kAccountId) >> account.account_id;
-        row.at(kDomainId) >> account.domain_id;
-        row.at("quorum") >> account.quorum;
-        row.at("data") >> account.json_data;
-        return account;
+
+        return makeAccount(result.at(0))
+            .match(
+                [](expected::Value<
+                    std::shared_ptr<shared_model::interface::Account>> &v) {
+                  return nonstd::make_optional(v.value);
```

Comment:
Would be nice to add that routine as a function, e.g:
```
Result.val() -> optional<Value>
Result.err() -> optional<Error>
```

Reply:
Also I think it possible to use generic ctor here `{v.value}` and `{}` for nullopt
Will be much easier to migrate to boost